File: implementation/transaction/src/main/java/com/financetracker/transaction/logic/model/OneTimeTransaction.java
package com.financetracker.transaction.logic.model;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Getter
@NoArgsConstructor
@Table(name = "one_time_transaction")
@Entity
public class OneTimeTransaction extends Transaction implements Transferable {

    private LocalDate date;

    @Embedded
    @AttributeOverride(name = "amount", column = @Column(name = "amount", precision = 27, scale = 6))
    private MonetaryAmount amount;

    @Builder(builderMethodName = "with")
    public OneTimeTransaction(final String id,
                              final String name,
                              final String description,
                              final Type type,
                              final Set<Label> labels,
                              final Transfer transfer,
                              final LocalDate date,
                              final MonetaryAmount amount) {
        super(id, name, description, type, labels, transfer);
        this.date = date;
        this.amount = amount;
    }
}

File: implementation/ui-financetracker/src/app/one-time-transaction/oneTimeTransaction.ts
import { MonetaryAmount } from "../bank-account/bankAccount";
import { Transfer, TransferStatus, Type } from '../transaction/transaction';

export class OneTimeTransaction {
    id: string;
    name: string;
    description: string;
    type: Type;
    labels: string[];
    transfer: Transfer;
    amount: MonetaryAmount;
    date: string;
    transferStatus: TransferStatus;

    constructor(id: string, name: string, description: string, type: Type, labels: string[], transfer: Transfer, amount: MonetaryAmount, date: string, transferStatus: TransferStatus) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.type = type;
        this.labels = labels;
        this.transfer = transfer;
        this.amount = amount;
        this.date = date;
        this.transferStatus = transferStatus;
    }
}
